invenio_records_rest/views.py

```diff
@@ -277,6 +299,73 @@ def need_record_permission_decorator(self, record, *args, **kwargs):
     return need_record_permission_builder
 
 
+class RecordsSuggestsResource(ContentNegotiatedMethodView):
+    """Resource for records suggests."""
+
+    view_name = '{0}_suggests'
+
+    def __init__(self, completion_opts,
+                 pid_type=None, pid_fetcher=None, read_permission_factory=None,
+                 create_permission_factory=None, search_index=None,
+                 search_type=None,
+                 search_serializers=None, default_media_type=None,
+                 item_links_factory=None, **kwargs):
+        """Constructor."""
+        super(RecordsSuggestsResource, self).__init__(
+            method_serializers={
+                'GET': search_serializers,
+            },
+            default_method_media_type={
+                'GET': default_media_type,
+            },
+            default_media_type=default_media_type,
+            **kwargs)
+        self.completion_opts = completion_opts
+        self.pid_type = pid_type
+        self.pid_fetcher = current_pidstore.fetchers[pid_fetcher]
+        self.read_permission_factory = read_permission_factory
+        self.create_permission_factory = create_permission_factory
+        self.search_index = search_index
+        self.search_type = search_type
+        self.item_links_factory = item_links_factory
+
+    def get(self, **kwargs):
+        """Search records.
+
+        :returns: the search result containing hits and aggregations as
+        returned by invenio-search.
+        """
+        text = request.values.get('text', "", type=str)
+
+        query = {
+            self.completion_opts['field']: {
+                'text': text,
+                'completion': self.completion_opts
+            }
+        }
+
+        # Execute search
+        search_result = current_search_client.suggest(
```

Comment:
Hmmm....actually this is not needed to begin with since we're only going to auto-complete from fully public vocabularies (perhaps we can post-pone it until we have to deal with search auto-completion).